### PR TITLE
Update .Net Core/SDK docs about version 2.0 to be about version 2.x

### PR DESCRIPTION
**Pull request by bergmeister**
## PR Summary

As per title. Due to the upgrade to .Net Core 2.1 in 6.1, it is better to not be specific about the version of .Net Core or the SDK.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [x] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Comment by bergmeister** [comment]
@TravisEz13 I addressed the spelling issues.

**Review by daxian-dbw — APPROVED** [review]

**Review by TravisEz13 — APPROVED** [review]

File: docs/building/macos.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 This guide supplements the [Linux instructions](./linux.md), as
 building on macOS is almost identical.
 
-.NET Core 2.0 (and by transitivity, us) only supports macOS 10.12+.
+.NET Core 2.x (and by transitivity, us) only supports macOS 10.12+.
```
Comment (dantraMSFT) [inline docs/building/macos.md]: Does the third bullet item also need to be changed?  

Downloads and installs a preview version of .NET Core SDK 2.0 to ~/.dotnet

File: docs/cmdlet-example/command-line-simple-example.md [hunk 2]
```diff
@@ -171,7 +171,7 @@ can't find the `netstandard.dll` "implementation" assembly for the version of th
 
 ### The fix for missing netstandard.dll
```
Comment (dantraMSFT) [inline docs/cmdlet-example/command-line-simple-example.md]: There are stale references to 2.0 in this section.  Two that stand out are

* "Install the .NET Core SDK 2.0 for Windows, if it isn't already installed."
This actually links to the .NET Core 2.1 page.  Perhaps updating the link label to also use 2.x

* Copy-Item 'C:\Program Files\dotnet\sdk\2.0.0\Microsoft...
A note about the version number varying might help so users arent' blindly copy/pasting the text and getting errors.

**Review by dantraMSFT — CHANGES_REQUESTED** [review]

**Comment by daxian-dbw** [comment]
@dantraMSFT I pushed a commit to address your comments. Please take another look.

**Review by dantraMSFT — APPROVED** [review]

**Comment by bergmeister** [comment]
@daxian-dbw Thanks for helping out, I was quite busy the last weeks due to moving house and forgot about this PR.

**Comment by daxian-dbw** [comment]
@bergmeister No worries, that's what we maintainers do 😄

**Pull request merged**